src/RemoteInboxNotifications/OptionRuleProcessor.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Rule processor that performs a comparison operation against an option value.
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Rule processor that performs a comparison operation against an option value.
+ */
+class OptionRuleProcessor implements RuleProcessorInterface {
+	/**
+	 * Performs a comparison operation against the option value.
+	 *
+	 * @param object $rule         The specific rule being processed by this rule processor.
+	 * @param object $stored_state Stored state.
+	 *
+	 * @return bool The result of the operation.
+	 */
+	public function process( $rule, $stored_state ) {
+		$option_value = get_option( $rule->option_name );
```

Comment:
I wonder if we should allow a default option to be passed here as part of the rule.

Reply:
It took me a while to get back to this, but that's a great idea :-) done in 6c3745395e89155a4ca55f2ee097e9bd916c4755